cbits/measure_off.c

```diff
@@ -17,19 +17,38 @@
 #include <stdatomic.h>
 #endif
 
+/*
+  Clang-6 does not enable proper -march flags for assembly modules
+  which leads to "error: instruction requires: AVX-512 ISA"
+  at the assembler phase.
+
+  Apple LLVM version 10.0.0 (clang-1000.11.45.5) is based on clang-6
+  https://en.wikipedia.org/wiki/Xcode#Toolchain_versions
+  and it's latest available version on macOS 10.13.
+
+  Disable AVX-512 instructions as they are most likely not supported
+  on the hardware running clang-6.
+*/
+#if (defined(__apple_build_version__) && __apple_build_version__ <= 10001145) \
+    || (defined(__clang_major__) && __clang_major__ <= 6)
+#define NO_AVX512
+#endif
+
+#if defined(__x86_64__) && !defined(NO_AVX512)
+#define USE_AVX512
```

Comment:
Thanks, @vshabanov, much appreciated! I have a tiny nit pick.

The name `USE_AVX512` is a bit misleading: it's not like we will definitely use AVX512 on any configuration, where `USE_AVX512` is defined. 

Could we please rework the PR to define `COMPILER_SUPPORTS_AVX512` macro and use it below as `#if defined(__x86_64__) && defined(COMPILER_SUPPORTS_AVX512)`?

Reply:
I've changed USE_AVX512 to COMPILER_SUPPORTS_AVX512.

Haven't done anything with #401 conflict yet as I don't know how to handle it better. If `__get_cpuid_count` is not supported then there's no point in having AVX512 functions as well. So we could either use `(__GNUC__ >= 6 || defined(__clang_major__))` to define COMPILER_SUPPORTS_AVX512 (which would make the macro badly named again) or we could just use it to `return false` in `has_avx512_vl_bw()`, which is probably fine.

Reply:
Second option please.

Reply:
Done. Merged #401.